crates/kit/src/run_ephemeral.rs

```diff
@@ -586,9 +586,9 @@ fn prepare_run_command_with_temp(
     // which would otherwise contain unreachable bridge DNS servers (e.g., 169.254.1.1).
     // Using --dns properly configures /etc/resolv.conf in the container.
     let host_dns_servers = read_host_dns_servers().or_else(|| {
-        // Fallback to public DNS if no usable DNS found in system configuration
-        // This ensures DNS works even when host has broken/unreachable DNS config
-        warn!("No usable DNS servers found in system configuration, falling back to public DNS (8.8.8.8, 1.1.1.1). This may not work in air-gapped environments.");
+        // Fallback to public DNS if no DNS servers found in system configuration
+        // This ensures DNS works in most cases, but may fail in air-gapped or VPN-only environments
+        debug!("No DNS servers found in system configuration, falling back to public DNS (8.8.8.8, 1.1.1.1)");
         Some(vec!["8.8.8.8".to_string(), "1.1.1.1".to_string()])
```

Comment:
I'm thinking we should actually never have this fallback. For one thing...it's a bit confusing for us to do in the case where there's no DNS servers at all in an offline system.

And if we somehow fail to parse DNS servers...I think it could be quite confusing to have a silent fallback. We really *shouldn't* hit this case anyways.